src/generator/AutoRest.Go/TemplateModels/ModelsTemplateModel.cs

```diff
@@ -100,6 +103,9 @@ public ModelsTemplateModel(ServiceClient serviceClient, string packageName)
                     {
                         PagedTypes.Add(m.ReturnValue().Body, m.NextLink());
                     }
+                    if (!m.NextMethodExists(serviceClient.Methods)) {
+                        NextMethodUndefined.Add(m.ReturnValue().Body);
+                    }
```

Comment:
>> With this changes, the generator no longer generates extra next operations.

don't generate extra preparer only when it is already defined in swagger. else generate always. I think you keep in account both cases. Right?